amstrax/plugins/radon_pulse_processing.py

```diff
@@ -0,0 +1,350 @@
+import numba
+import numpy as np
+import strax
+from immutabledict import immutabledict
+
+import amstrax
+
+export, __all__ = strax.exporter()
+__all__ += ['NO_PULSE_COUNTS']
+
+# These are also needed in peaklets, since hitfinding is repeated -> then import with the plugin hitself
+HITFINDER_OPTIONS = tuple([
+    strax.Option(
+        'hit_min_amplitude',
+        default='pmt_commissioning_initial',
+        help='Minimum hit amplitude in ADC counts above baseline. '
+             'See straxen.hit_min_amplitude for options.'
+    ),
+    strax.Option(
+        'hit_threshold',
+        default=10,
+        help='Hitfinder threshold in ADC counts above baseline'
+        )
+    ])
+
+
+@export
+@strax.takes_config(
+    strax.Option(
+        'baseline_samples',
+        default=40, infer_type=False,
+        help='Number of samples to use at the start of the pulse to determine '
+             'the baseline'),
+    # PMT pulse processing options
+    strax.Option(
+        'pmt_pulse_filter',
+        default=None, infer_type=False,
+        help='Linear filter to apply to pulses, will be normalized.'),   
+    strax.Option(
+        'save_outside_hits',
+        default=(3, 20), infer_type=False,
+        help='Save (left, right) samples besides hits; cut the rest'),
+    strax.Option(
+        'n_tpc_pmts', type=int,
+        help='Number of TPC PMTs'),
+    strax.Option(
+        'check_raw_record_overlaps',
+        default=False, track=False, infer_type=False,
+        help='Crash if any of the pulses in raw_records overlap with others '
+             'in the same channel'),
```

Comment:
```suggestion
    strax.Option(
        'check_raw_record_overlaps',
        default=True, track=False, infer_type=False,
        help='Crash if any of the pulses in raw_records overlap with others '
             'in the same channel'),
```
Maybe we want to turn this on by default?

Reply:
Good idea